plugins/modules/azure_rm_aks.py

```diff
@@ -630,10 +691,28 @@ def compare_addon(origin, patch, config):
                             if profile_result['name'] == profile_self['name']:
                                 matched = True
                                 os_disk_size_gb = profile_self.get('os_disk_size_gb') or profile_result['os_disk_size_gb']
-                                if profile_result['count'] != profile_self['count'] \
-                                        or profile_result['vm_size'] != profile_self['vm_size'] \
-                                        or profile_result['os_disk_size_gb'] != os_disk_size_gb \
-                                        or profile_result['vnet_subnet_id'] != profile_self.get('vnet_subnet_id', profile_result['vnet_subnet_id']):
+                                if profile_result['count'] != profile_self['count'] and profile_self['count'] :
+                                    self.log(("Agent Profile Diff - Origin {0} / Update {1}".format(str(profile_result), str(profile_self))))
+                                    to_be_updated = True
+                                elif profile_result['vm_size'] != profile_self['vm_size'] and profile_self['vm_size'] :
+                                    self.log(("Agent Profile Diff - Origin {0} / Update {1}".format(str(profile_result), str(profile_self))))
+                                    to_be_updated = True
+                                elif profile_result['os_disk_size_gb'] != os_disk_size_gb and os_disk_size_gb :
+                                    self.log(("Agent Profile Diff - Origin {0} / Update {1}".format(str(profile_result), str(profile_self))))
+                                    to_be_updated = True
+                                elif profile_result['vnet_subnet_id'] != profile_self.get('vnet_subnet_id', profile_result['vnet_subnet_id']) and profile_self.get('vnet_subnet_id', profile_result['vnet_subnet_id']) :
+                                    self.log(("Agent Profile Diff - Origin {0} / Update {1}".format(str(profile_result), str(profile_self))))
+                                    to_be_updated = True
+                                elif profile_result['type'] != profile_self['type'] and profile_self['type'] :
```

Comment:
make sure type is updatable, otherwise we should remove it

Reply:
Thanks for your review! I will retest and check it!